packages/Makefile

```diff
@@ -0,0 +1,45 @@
+# Figure out which Zarf binary we should use based on the operating system we are on
+ZARF_BIN := zarf
+UNAME_S := $(shell uname -s)
+UNAME_P := $(shell uname -p)
+ifneq ($(UNAME_S),Linux)
+	ifeq ($(UNAME_S),Darwin)
+		ZARF_BIN := $(addsuffix -mac,$(ZARF_BIN))
+	endif
+	ifeq ($(UNAME_P),i386)
+		ZARF_BIN := $(addsuffix -intel,$(ZARF_BIN))
+	endif
+	ifeq ($(UNAME_P),arm64)
+		ZARF_BIN := $(addsuffix -apple,$(ZARF_BIN))
+	endif
+endif
+
+PACKAGE_PATH :=  $(shell cd $(name) && pwd)
+
+.PHONY: default 
+default: help
+
+.PHONY: package
+package:
+ifdef name
+	@cd $(name) && $(ZARF_BIN) package create --confirm 
```

Comment:
This only current works if $(ZARF_BIN) is somewhere in $PATH. I don't think we make that assumption in our other Makefiles.

Reply:
No and that can change now that the pr is done so I’ll update that as well.

Reply:
As soon as we update the zarf version it can be done the way the others are.

Reply:
One issue I have is in the examples (which this wouldn't be) we assume they have cloned the repo. If we are building composable packages we should be assuming they are using a release rather than using the github repo which uses another make command to build and place zarf into a created folder called ../sync

Reply:
but it only does that if you use the make command within the examples folder otherwise it places it in build which is relative to the path that exists for the repo and not anything related to the os. I think its a way more resonable assumption that if a user wants to use zarf to build composed packages then they would have zarf. I am open to other suggestions though.

Reply:
Similar to how when running helm it is expected that it exists within the os path otherwise it will not work.

Reply:
Take a look at the way I set it up in the zarf-package-software-factory repo. Not saying that's the way we always have to do it but it should give some ideas. In there I just added documentation that having `zarf` installed and present in PATH a prerequisite.

Reply:
@jeff-mccoy which if fair, but zarf isn't native to itself unless itself exists. 

I did add it to the pre-requisites in the readme. 

I also don't think that this will be the way we build them long term. This was just a way to provide the initial behavior without relying on a registry like artifact hub at this point. I think we could eliminate most of this once that is the case. Though that will still have the pre-requisite that zarf exists and you are able to use it regardless of where it exists in its environment.

Reply:
I can definitely see us leveraging OCI Artifact spec for this in the future.

Reply:
Yeah, I think if you run it how it is it definitely shows that idea of replacing locally existing packages with a sha and url. Which I’m hoping will be possible as a more useful mvp than what I currently have. Because then those actions could always change to point to any package registry.